src/pd/client.rs

```diff
@@ -266,6 +267,58 @@ impl<KvC: KvConnect + Send + Sync + 'static, Cl> PdRpcClient<KvC, Cl> {
     }
 }
 
+/// This client wraps a PdClient and encodes/decodes keys and regions according to
+/// TiKV transaction codec.
+pub struct PdCodecClient<PdClient = PdRpcClient> {
```

Comment:
I don't think we need a separate client for this, you can just add the encoding to the existing client.